### ISSUE TITLE
List supported Elliptic Curves

### ISSUE THREAD

**Issue by J08nY** [randombit/botan#1338]
There is currently no way to enumerate supported Elliptic Curves, apart from trying some known OIDs in `OIDS::lookup`. This would be quite useful. (list their string names as Botan uses them and also their OIDs)

I could do a PR if this feature would be approved, however currently I am not quite sure for a right place for such enumeration.

**Comment by randombit** [comment]
This would be useful, especially over time as groups are added and removed.

Probably best spot is in `ec_named.cpp`, add new static function

```
std::vector<std::string> EC_Group::known_ec_groups()
   {
   return {
      "secp256r1",
        ...
       };
    }
```

There are some refactoring planned for how EC_Groups are stored so this implementation will eventually change, but I could handle that with the rest of the refactoring. The above would do for now.

Someone who needs the OID can call `OIDS:lookup` on the returned strings.

If you do this please also add test that verifies that for each value returned by `known_ec_groups` we are also able to retrieve the representation as well as lookup an OID for that name.

Bonus points for updating `ECC_Randomized_Tests` (`tests/unit_ecc.cpp`) to use this function instead of hardcoding its own list of curves.

**Comment by randombit** [comment]
Hm, things to contemplate:

Does a `std::set` make more more sense? There is no meaning to the ordering of the return value.

Is it useful to cache the value (as a static const) and return a const reference, instead of new object?

**Comment by J08nY** [comment]
Right, both things make sense I think. Will work on the PR soon.

**Comment by randombit** [comment]
Fixed in #1338 thanks!
